### PR TITLE
Use chrony for time synchronization

### PR DESCRIPTION
**Pull request by smithfarm**

### CODE REVIEW

**Comment by smithfarm** [comment]
Note that this is a little polluted with xfsprogs changes that would be better dealt with in the image generation script. I'm doing it this way due to "reasons".

**Comment by smithfarm** [comment]
This has been tested and does what it's supposed to do.

File: teuthology/nuke/actions.py [hunk 1]
```diff
@@ -379,13 +379,16 @@ def synch_clocks(remotes):
     for remote in remotes:
         remote.run(
             args=[
-                'sudo', 'service', 'ntp', 'stop',
+                'sudo', 'service', 'ntp', 'stop', run.Raw('||'),
+                'sudo', 'service', 'chronyd', 'stop',
                 run.Raw('&&'),
-                'sudo', 'ntpd', '-gq',
+                'sudo', 'ntpd', '-gq', run.Raw('||'),
+                'sudo', 'chronyc', 'sources', run.Raw('||'),
```
Comment (kshtsk) [inline teuthology/nuke/actions.py]: The '||' is extra in this place, isn't it?

File: teuthology/task/clock.py [hunk 2]
```diff
@@ -35,14 +35,18 @@ def task(ctx, config):
         rem.run(
             args = [
                 'sudo', 'service', 'ntp', 'stop', run.Raw('||'),
-                'sudo', 'systemctl', 'stop', 'ntp.service',
+                'sudo', 'systemctl', 'stop', 'ntp.service', run.Raw('||'),
```
Comment (kshtsk) [inline teuthology/task/clock.py]: Don't `service nap stop` and `systemctl stop ntp.service` do the same?

Comment (smithfarm) [reply teuthology/task/clock.py]: Yes, I'm inclined to drop the `service . . . stop` bits.

**Comment by smithfarm** [comment]
I changed the code that stops and starts the `ntpd.service` or `chronyd.service` units, but we don't run this code in our testing, so it's untested.

**Pull request merged**